skore/tests/unit/sklearn/train_test_split/test_train_test_split.py

```diff
@@ -180,3 +180,41 @@ def test_train_test_split_kwargs():
     output2 = train_test_split(X=X, y=y, random_state=0)
 
     assert output1 == output2
+
+
+def test_train_test_split_dict_kwargs():
+    """Passing data without keyword arguments with return_dict=True
+    should raise ValueError."""
+
+    X = [[1]] * 20
+    y = [0] * 10 + [1] * 10
+
+    with pytest.raises(ValueError):
+        train_test_split(X, y, random_state=0, return_dict=True)
+
+
+def test_train_test_split_check_dict():
+    """Check dict is returned if `return_dict` argument is True"""
+    X = [[1]] * 20
+    y = [0] * 10 + [1] * 10
+    output = train_test_split(X=X, y=y, random_state=0, return_dict=True)
+    assert type(output) is dict
+
+
+def test_train_test_split_check_dict_unsupervised_case():
+    """Check 2 keys are is returned if `return_dict` argument is True for
+    unsupervised case."""
+
+    X = [[1]] * 20
+    output = train_test_split(X=X, random_state=0, return_dict=True)
+    assert len(output.keys()) == 2
+
+
+def test_train_test_split_check_dict_no_X_no_y():
+    """Check if a dict is returned when X, y are None and a third kwarg is passed."""
+
+    z = [[1]] * 20
+    output = train_test_split(z=z, test_size=0.33, random_state=0, return_dict=True)
+    keys = output.keys()
+    assert len(keys) == 2
+    assert list(keys) == ["z_train", "z_test"]
```

Comment:
Since we check the list, checking the length is redundant
```suggestion
    keys = output.keys()
    assert list(keys) == ["z_train", "z_test"]
```

Reply:
Added in https://github.com/probabl-ai/skore/pull/1460/commits/9b1a875b869f49ed0bcccdd80777c19502345226